src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -1424,4 +1428,15 @@ export default class DefaultDeviceController implements DeviceControllerBasedMed
     }
     this.logger.info(s);
   }
+
+  private static isMediaDeviceInfo(device: unknown): device is MediaDeviceInfo {
+    return (
+      !!device &&
+      typeof device === 'object' &&
+      'deviceId' in device &&
+      'groupId' in device &&
+      'kind' in device &&
+      'label' in device
+    );
+  }
```

Comment:
Make this a free function rather than a static method.

You also don't need `!!` here.

Reply:
Updated. But I followed the example here https://github.com/aws/amazon-chime-sdk-js/blob/master/src/devicecontroller/VideoTransformDevice.ts#L44. Do we need !! there?